api/budget/views.py

```
@@ -15,3 +27,57 @@ def get_queryset(self):
         return Budget.objects.select_related("scenario").filter(
             scenario__account=self.request.user.iaso_profile.account
         )
+
+    def get_serializer_class(self):
+        if self.action == "create":
+            return BudgetCreateSerializer
+        return BudgetSerializer
+
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        scenario = serializer.validated_data["scenario"]
+
+        # TODO: Years should come from the frontend?
+        start_year = 2024
```

Years are on the scenario now, you could use that instead, if not set on the scenario, I would say that we can return an error specifying that this is missing